config/fcos/v1_5_exp/translate_test.go

```diff
@@ -219,6 +219,73 @@ func TestTranslateBootDevice(t *testing.T) {
 			},
 			report.Report{},
 		},
+		// LUKS, x86_64, with Tang set for offline provisioning
+		{
+			Config{
+				BootDevice: BootDevice{
+					Luks: BootDeviceLuks{
+						Tang: []base.Tang{{
+							URL:           "https://example.com/",
+							Thumbprint:    util.StrToPtr("z"),
+							Advertisement: util.StrToPtr("{\"payload\": \"xyzzy\"}"),
+						}},
+					},
+				},
+			},
+			types.Config{
+				Ignition: types.Ignition{
+					Version: "3.4.0-experimental",
+				},
+				Storage: types.Storage{
+					Luks: []types.Luks{
+						{
+							Clevis: types.Clevis{
+								Tang: []types.Tang{{
+									URL:           "https://example.com/",
+									Thumbprint:    util.StrToPtr("z"),
+									Advertisement: util.StrToPtr("{\"payload\": \"xyzzy\"}"),
+								}},
+							},
+							Device:     util.StrToPtr("/dev/disk/by-partlabel/root"),
+							Label:      util.StrToPtr("luks-root"),
+							Name:       "root",
+							WipeVolume: util.BoolToPtr(true),
+						},
+					},
+					Filesystems: []types.Filesystem{
+						{
+							Device:         "/dev/mapper/root",
+							Format:         util.StrToPtr("xfs"),
+							Label:          util.StrToPtr("root"),
+							WipeFilesystem: util.BoolToPtr(true),
+						},
+					},
+				},
+			},
+			[]translate.Translation{
+				{From: path.New("yaml", "version"), To: path.New("json", "ignition", "version")},
+				{From: path.New("yaml", "boot_device", "luks", "tang", 0, "url"), To: path.New("json", "storage", "luks", 0, "clevis", "tang", 0, "url")},
+				{From: path.New("yaml", "boot_device", "luks", "tang", 0, "thumbprint"), To: path.New("json", "storage", "luks", 0, "clevis", "tang", 0, "thumbprint")},
+				{From: path.New("yaml", "boot_device", "luks", "tang", 0, ""), To: path.New("json", "storage", "luks", 0, "clevis", "tang", 0, "advertisement")},
```

Comment:
I am surprised by this, I expected `{From:....., "tang",0,"advertisement"} `but instead I got `{From:....., "tang",0,""} `
Is that an indication of something wrong? @bgilbert

Reply:
Yes, that shouldn't be happening.  It's because you're tagging the `Advertisement` field in `schema.go` with `json:` instead of `yaml:`.